tlatools/org.lamport.tlatools/test/tla2sany/SanyTranslator.java

```diff
@@ -0,0 +1,1633 @@
+package tla2sany;
+
+import tla2sany.modanalyzer.SyntaxTreePrinter;
+import tla2sany.parser.SyntaxTreeNode;
+import tla2sany.parser.TLAplusParser;
+import tla2sany.parser.TLAplusParserConstants;
+import tla2sany.st.SyntaxTreeConstants;
+import tla2sany.AstNode.Kind;
+
+import java.io.PrintWriter;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.Assert;
+
+/**
+ * Handles the translation of raw SANY output to the test file AST DSL.
+ */
+public class SanyTranslator {	
+
+	/**
+	 * Sometimes SANY provides very little structure in its parse output,
+	 * basically just a flat list of tokens. This is most common when dealing
+	 * with constructs that SANY doesn't care much about, like proof-related
```

Comment:
How much is missing in SANY to add more structure for proof-related syntax?  There is an unfinished attempt to replace TLAPS' parser with SANY and this sounds like a potential show-stopper for this effort.

Reply:
Not really that much. I would say use-or-hide is the main culprit, which you can see because I had to break that parsing logic out into its own set of functions instead of writing it in the main switch statement. It's annoying but surmountable. The parse output of SANY could also be improved.

Reply:
Makes sense, I suggest adding this note to the code.

Reply:
Added this note as a comment in the code here.